client/tests/components/form/MultiStepFormBase.test.tsx

```diff
@@ -0,0 +1,275 @@
+    // these are tested more fully in the MultiStepFormButtons test
+    render(<MultiStepFormBase {...defaultProps} disabled={false} />);
+    const saveAndContinueButton = screen.getByRole("button", {
+      name: /Save and Continue/i,
+    });
+    expect(screen.getByLabelText(/field1*/i)).toHaveValue("test field1");
+    expect(screen.getByRole("button", { name: /Back/i })).toBeDisabled();
+    expect(screen.getByRole("button", { name: /Cancel/i })).not.toBeDisabled();
+    expect(saveAndContinueButton).not.toBeDisabled();
+    await fireEvent.click(saveAndContinueButton);
+    expect(mockOnSubmit).toHaveBeenCalled();
+  });
+
+  it("navigation and submit buttons work on second form page", async () => {
+    useParams.mockReturnValue({
+      formSection: "2",
+      operation: "025328a0-f9e8-4e1a-888d-aa192cb053db",
+    } as QueryParams);
+
+    render(<MultiStepFormBase {...defaultProps} disabled={false} />);
+    const saveAndContinueButton = screen.getByRole("button", {
+      name: /Save and Continue/i,
+    });
+    expect(screen.getByLabelText(/field2*/i)).toHaveValue("test field2");
+    expect(screen.getByRole("button", { name: /Back/i })).not.toBeDisabled();
+    expect(screen.getByRole("button", { name: /Cancel/i })).not.toBeDisabled();
+    expect(saveAndContinueButton).not.toBeDisabled();
+    await fireEvent.click(saveAndContinueButton);
+    expect(mockOnSubmit).toHaveBeenCalled();
+  });
+
+  it("navigation and submit buttons work on last form page", async () => {
+    useParams.mockReturnValue({
+      formSection: "3",
+      operation: "025328a0-f9e8-4e1a-888d-aa192cb053db",
+    } as QueryParams);
+    render(<MultiStepFormBase {...defaultProps} disabled={false} />);
+    expect(screen.getByLabelText(/field3*/i)).toHaveValue("test field3");
+    const submitButton = screen.getByRole("button", {
+      name: /test submit button text/i,
+    });
+
+    expect(screen.getByRole("button", { name: /Back/i })).not.toBeDisabled();
+    expect(screen.getByRole("button", { name: /Cancel/i })).not.toBeDisabled();
+    expect(
+      screen.getByRole("button", { name: /test submit button text/i }),
+    ).not.toBeDisabled();
+    await fireEvent.click(submitButton);
+    expect(mockOnSubmit).toHaveBeenCalled();
+  });
+
+  it("submission is disabled if form is still submitting", async () => {
+    useParams.mockReturnValue({
+      formSection: "1",
+      operation: "create",
+    } as QueryParams);
+    // let resolve: (v: unknown) => void;
+    // const mockOnSubmit = vi.fn().mockReturnValue(
+    //   new Promise((_resolve) => {
+    //     resolve = _resolve;
+    //   })
+    // );
+    render(<MultiStepFormBase {...defaultProps} disabled={false} />);
+    const saveAndContinueButton = screen.getByRole("button", {
+      name: /Save and Continue/i,
+    });
+    await fireEvent.click(saveAndContinueButton);
+    await waitFor(() => {
+      expect(saveAndContinueButton).toBeDisabled();
```

Comment:
I wasn't able to get this working. This was the best example I found so far but it didn't work: https://github.com/orgs/react-hook-form/discussions/8866

Reply:
I've used `tobeDisabled` as well; I think the issue here is timing. Have you come across good alternatives to `waitFor`?